pep-0615.rst

```diff
@@ -0,0 +1,560 @@
+It will be designed to be compatible with both newer ``importlib.resources`` [importlib_resources] access patterns and older access patterns like ``pkgutil.get_data`` [pkgutil_data] .
+
+While it is designed explicitly for the use of CPython,
+the ``tzdata`` package is intended as a public package in its own right,
+and it may be used as an "official" source of time zone data for Python packages.
+
+.. _search-path-config:
+
+Search path configuration
+-------------------------
+
+The time zone search path is very system-dependent, and sometimes even application-dependent,
+and as such it makes sense to provide options to customize it.
+This PEP provides for three such avenues for customization:
+
+1. Global configuration via compile-time options
+2. Per-run configuration via an environment variable
+3. Runtime configuration change via a ``set_tzpath`` function
+
+Compile-time options
+####################
+
+It is most likely that downstream distributors will know exactly where their system time zone data is deployed,
+and so a compile-time option ``PYTHONTZPATH`` will be provided to set the default search path.
+
+The ``PYTHONTZPATH`` option should be a string delimited by ``os.pathsep``,
+listing possible locations for the time zone data to be deployed (e.g. ``/usr/share/zoneinfo``).
+
+Environment variables
+#####################
+
+When initializing the ``TZPATH`` (and whenever ``set_tzpath`` is called with no arguments),
+the ``zoneinfo`` module will use two environment variables,
+``PYTHONTZPATH`` and ``PYTHONTZPATH_APPEND``, if they exist, to set the search path.
+
+Both are ``os.path`` delimited strings.
+``PYTHONTZPATH`` *replaces* the default time zone path,
+whereas ``PYTHONTZPATH_APPEND`` appends to the end of the time zone path.
+
+Some examples of the proposed semantics::
+
+    $ python print_tzpath.py
+    ("/usr/share/zoneinfo",
+     "/usr/lib/zoneinfo",
+     "/usr/share/lib/zoneinfo",
+     "/etc/zoneinfo")
+
+    $ PYTHONTZPATH="/etc/zoneinfo:/usr/share/zoneinfo" python print_tzpath.py
+    ("/etc/zoneinfo",
+     "/usr/share/zoneinfo")
+
+    $ PYTHONTZPATH="" python print_tzpath.py
+    ()
+
+    $ PYTHONTZPATH_APPEND="/my/directory" python print_tzpath.py
+    ("/usr/share/zoneinfo",
+     "/usr/lib/zoneinfo",
+     "/usr/share/lib/zoneinfo",
+     "/etc/zoneinfo",
+     "/my/directory")
+
+``set_tzpath`` function
+#######################
+
+``zoneinfo`` provides a ``set_tzpath`` function that allows for changing the search path at runtime.
+
+.. code-block:: python3
+
+    def set_tzpath(tzpaths: Optional[Sequence[Union[str, Pathlike]]]) -> None:
+        ...
+
+When called with a sequence of paths,
+this function sets ``zoneinfo.TZPATH`` to a tuple constructed from the desired value.
+When called with no arguments or ``None``,
+this function resets ``zoneinfo.TZPATH`` to the default configuration.
+
+This is likely to be primarily useful for (permanently or temporarily)
```

Comment:
Hm.. I will try to address this tonight.

To be honest, part of the reason is basically that I want to give an officially sanctioned way to do it because I suspect people will just start messing around with ``TZPATH`` even if I make it a tuple or something.

If we later decide that `set_tzpath` should affect cache behavior, or have some other sort of side effect, it will be a lot easier to cause `set_tzpath` to have an effect than it will to have assigning `zoneinfo.TZPATH` to something new.

I mainly found it useful for testing to force the use of `tzdata` in a programmatic way. I imagine other people will do something similar to hack around limited "bad data" zones or something, like:

```python
def get_zone_from_tzdata(key):
    old_tzpath = zoneinfo.TZPATH
    zoneinfo.TZPATH = ()
    rv = zoneinfo.ZoneInfo(key)
    zoneinfo.TZPATH = old_tzpath
    return rv
```

So, I'd rather they use an API I've specifically designed than one they've hacked together.

Basically, my reasoning is less, "This is something you should do" and more "I know you're going to do it anyway and I want to exercise *some* control over it." I'll think how best to capture that without explicitly making the analogy to parents who let their kids have a house party because, "They're going to drink anyway, and we'd rather they do it at home where we know they are safe." 😅.

Reply:
OK, I've added some minor justification here. Hard to really justify this because I'm sort of on the fence about including it at all, but I suppose we can kick that question to the actual discussion stage of the process.